File: bench/benchmarks/trial_primes/trial_primes.go
//go:build ignore

package main

import "fmt"

func main() {
	N := int64(200000)
	count := int64(0)
	for n := int64(2); n <= N; n++ {
		isPrime := true
		for d := int64(2); d*d <= n; d++ {
			if n%d == 0 {
				isPrime = false
				break
			}
		}
		if isPrime {
			count++
		}
	}
	fmt.Println(count)
}

File: bench/benchmarks/trial_primes/trial_primes.c
/* Benchmark: count primes up to N by trial division. Same algorithm as .monk. */
#include <stdio.h>
#include <stdbool.h>

int main(void) {
    long N = 200000;
    long count = 0;
    for (long n = 2; n <= N; n++) {
        bool is_prime = true;
        for (long d = 2; d * d <= n; d++) {
            if (n % d == 0) {
                is_prime = false;
                break;
            }
        }
        if (is_prime) count++;
    }
    printf("%ld\n", count);
    return 0;
}
